lib/browser/api/net.js

```diff
@@ -41,7 +41,46 @@ class IncomingMessage extends Readable {
   }
 
   get headers () {
-    return this.urlRequest.rawResponseHeaders
+    // list of headers that Node.js discards duplicates for
+    // see https://nodejs.org/api/http.html#http_message_headers
+    const discardableDuplicateHeaders = [
+      'content-type',
+      'content-length',
+      'user-agent',
+      'referer',
+      'host',
+      'authorization',
+      'proxy-authorization',
+      'if-modified-since',
+      'if-unmodified-since',
+      'from',
+      'location',
+      'max-forwards',
+      'retry-after',
+      'etag',
+      'last-modified',
+      'server',
+      'age',
+      'expires'
+    ]
+
+    const filteredHeaders = {}
+    const rawHeaders = this.urlRequest.rawResponseHeaders
+    Object.keys(rawHeaders).forEach(header => {
+      if (header in filteredHeaders && discardableDuplicateHeaders.includes(header)) {
```

Comment:
do we need to `toLowerCase` here?

Reply:
yes, the headers aren't always consistent so it's simplest to lowercase them all